Tests/Marketplace/search_and_uninstall_pack.py

```diff
@@ -18,6 +19,126 @@
+
+    """
+    try:
+        for attempt in range(attempts_count - 1, -1, -1):
+            try:
+                if should_try_handler and not should_try_handler():
+                    # if the method exist and we should not try again.
+                    return True, None
+
+                # should_try_handler return True, we are trying to send request.
+                logging.info(f"{prior_message}, Attempt: {attempts_count - attempt}/{attempts_count}")
+                response, status_code, headers = demisto_client.generic_request_func(client,
+                                                                                     path=path,
+                                                                                     method=method,
+                                                                                     accept=accept,
+                                                                                     _request_timeout=request_timeout)
+
+                if 200 <= status_code < 300 and status_code != 204:
+                    if success_handler:
+                        # We have a method to run as we were returned a success status code.
+                        return success_handler(response)
+
+                    # No handler, just return True.
+                    return True, None
+
+                else:
+                    logging.info(f"Got bad response for updating status: {response}")
+
+                if not attempt:
+                    # No attempts left, raise an exception that the request failed.
+                    raise Exception(f"Got bad status code: {status_code}, headers: {headers}")
+
+                logging.warning(f"Got bad status code: {status_code} from the server, headers: {headers}")
+
+            except ApiException as ex:
+                if api_exception_handler:
+                    api_exception_handler(ex)
+                if not attempt:  # exhausted all attempts, understand what happened and exit.
+                    raise Exception(f"Got status {ex.status} from server, message: {ex.body}, headers: {ex.headers}") from ex
+                logging.debug(f"Process failed, got error {ex}")
+            except (HTTPError, HTTPWarning) as http_ex:
+                if http_exception_handler:
+                    http_exception_handler(http_ex)
+                if not attempt:  # exhausted all attempts, understand what happened and exit.
+                    raise Exception("Failed to perform http request to the server") from http_ex
+                logging.debug(f"Process failed, got error {http_ex}")
+
+            # There are more attempts available, sleep and retry.
+            logging.debug(f"{retries_message}, sleeping for {sleep_interval} seconds.")
+            sleep(sleep_interval)
+    except Exception as e:
+        logging.exception(f'{exception_message}. Additional info: {str(e)}')
```

Comment:
we are not raising the exception on purpose?

Reply:
Yes, tried to keep the implementation we had before. It does fail the build itself as we log exception.